### ISSUE TITLE
A fix of `@stylistic/selector-pseudo-class-parentheses-space-inside` writes a descendant combinator into a selector whose comment stands in front of a closing parenthesis

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#161]
Found while fixing [#127](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/127), and unrelated to it: this needs no empty argument and reproduces on `main` as it stands.

### What minimal example or steps are needed to reproduce the bug?

```css
a:not( /*c*/ ):is(b) {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-pseudo-class-parentheses-space-inside": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

```css
a:not(/*c*/):is(b) {}
```

The whitespace inside the parentheses of `:not()` is what the option asks about, and it is the only thing a fix has any business touching.

### What actually happened?

```css
a:not(/*c*/) :is(b) {}
```

A space appears between `)` and `:is`, and there was no warning about it. The selector no longer means what it meant: a compound selector has become a descendant combination, so the rule matches different elements after `--fix` than before it.

Both options do it, each writing its own whitespace inside the parentheses and the same space outside them:

```css
/* never  */ a:not( /*c*/ ):is(b) {}   →   a:not(/*c*/) :is(b) {}
/* always */ a:not( /*c*/ ):is(b) {}   →   a:not( /*c*/ ) :is( b ) {}
```

The fix does not settle either, where both pseudo-classes carry a comment. Under `always`:

```css
a:not(/*c*/):is(/*c*/) {}     /* the file */
a:not( /*c*/ ):is( /*c*/ ) {} /* after one run */
a:not( /*c*/ ) :is( /*c*/ ) {} /* after another, and the combinator is in */
```

The second run reports nothing new — it is the first run's own warning, fixed a second time, that moves the code.

### Do you have a proposal to fix the bug?

The rule reads a selector with `parseSelector`, writes the whitespace onto the nodes of the tree and puts `String(selectorTree)` back into the rule node. That last step is only sound while the parser gives the selector back the way it took it, and a comment standing in front of a closing parenthesis is one of the places where it does not:

```js
parser((root) => root).processSync(`a:not( /*c*/ ):is(b)`)   // 'a:not( /*c*/) :is(b)'
```

The space before `)` is dropped — it is not in the tree at all, which is [#128](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/128) — and a space of the same width turns up in front of `:is` instead, since the comment is folded into the raws of the node before it and the run behind it is filed as that node's trailing space. So every fix this rule makes in such a selector, in whichever pseudo-class of it, carries the moved space into the file.

The measure that suits this is the one `selector-descendant-combinator-no-non-space` takes for [#159](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/159): report nothing at all in a selector the parser does not give back as it took it, since neither the position of a warning nor the selector a fix would write back can be trusted there. #127 has already put a narrow form of that guard in this rule, keyed on a pseudo-class whose argument holds no node, and widening it to every mismatch is one line.

The cost is what makes it a decision rather than a change: the mismatch a comment before a closing parenthesis produces is exactly the shape #128 is about, and passing such a selector over would silence #128's eternal warning by taking away the fix that goes with it. That is one of the two answers #128 weighs, so the two want settling together.

**Comment by firefoxic** [comment]
A comma does the same as a closing parenthesis, and the symptom it leaves is a deletion rather than a combinator. The whitespace between a comment and the comma behind it is not in the tree either:

```js
parser((root) => root).processSync(`a:not(b, /*c*/ , c)`)   // 'a:not(b, /*c*/, c)'
```

so any fix this rule makes in such a selector takes that space away. Both options reach it, each on the shape its own fix answers:

```css
/* always */ a:not(b, /*c*/ , c) {}     →   a:not( b, /*c*/, c ) {}
/* never  */ a:not( b, /*c*/ , c ) {}   →   a:not(b, /*c*/, c) {}
```

Nothing is reported about that space, before or after. It needs no empty argument and reproduces on `main` as it stands, so it is the same defect seen from another side rather than a second one — the trigger is an argument whose last node is a comment, and what follows the comment decides which of the two symptoms the fix leaves. An argument holding a node in front of the comment keeps its whitespace and is unaffected, `a:not(b /*c*/ , c)` for one.

Whatever answers this issue has to answer both.
